en/docs/references/ballerina-expression-syntax.md

```diff
@@ -1,422 +1,485 @@
+HTTP cookies keep stateful information against a client. You can use the `request` variable's `getCookies` function to read cookies present in the request as an array.
 
-#### Examples
+###### Example:
 
-- Reading the value of an HTTP header (`Access-Control-Allow-Origin`) passing the header name.
+To read all the cookies available in the request, use the following expression:
 
   ```ballerina
-  check request.getHeader("Access-Control-Allow-Origin")
-  
+  request.getCookies()
   ```
-  
-- Reading the value of an HTTP header by passing a variable that contains the header name (`headerName`). Note the omission of the double-quotes.
 
-  ```ballerina
-  check request.getHeader(headerName)
-  ```
+## JSON
 
-### Checking whether a header is present in the request
+JSON is a textual format for representing a collection of values: a simple value, an array of values, or an object. Ballerina has a single type named `json` that can represent any JSON value. Thus, `json` is a built-in union type in Ballerina that can contain values of the types `nil` (as the null value), `boolean`,` int`,` float`, `decimal`, `string`, `json[]`, and `map<json>`.
 
-Applications sometimes need to check the availability of a header before proceeding to something like reading the header value. You can use the  `hasHeader`  function in the `request` object for this purpose.
+### Simple values in JSON 
 
-#### Examples
+Ballerina has a single type named `json` that can represent any JSON value. You can use `json` which is a built-in union type in Ballerina, to store values of the types `nil` (as the null value), `boolean`, `int`, `float`, `decimal`, `string`, `json[]`, and `map<json>`. 
 
-- Checking if a particular HTTP header (`content-length`) exists given the header name.
+###### Examples:
 
-  ```ballerina
-  request.hasHeader("content-length")
-  ```
-  
-- Checking if a particular HTTP header (`content-length`) exists by passing a variable (`contentLength`) containing the header name. Note the omission of the double-quotes.
+To initialize a `json` type variable, you can use the following:
+```ballerina
+    "Alex"
 
-  ```ballerina
-  request.hasHeader(contentLength)
-  ```
+    5.36
+    
+    false
+    
+    [1, false, null, "foo", {first: "Alex", last: "John"}]
+    
+    {name: "Alex", age: null, marks: {math: 90, language: 95}}
+    
+     null
+```
 
-### Reading the JSON Payload of the request
+### JSON Objects
 
-JSON is a common content type used for HTTP communication. The `request` variable’s `getJsonPayload` function can be used to read the JSON payload sent with the HTTP request. Note that the `getJsonPayload` function’s return type is `json|ClientError`.
+Ballerina represents JSON objects as maps. You can define a JSON object to hold values of any `json` type (`string`, `number`, `boolean`, `object`, `array`, or `null`). A JSON object can be declared in two ways:
+    -  Using the variable type `json`.
```

Comment:
`json` is not just used for JSON objects. The following is also valid.

```ballerina
json j = 1;
```

Wonder if this could cause confusion?